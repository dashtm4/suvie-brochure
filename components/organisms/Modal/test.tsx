import { shallow } from 'enzyme'
import toJSON from 'enzyme-to-json'

import { Modal } from '.'

describe('<Modal />', () => {
  it('should renders correct <Modal />', () => {
    const toggleModal = jest.fn()
    const children = 'test'

    const wrapper = shallow(<Modal toggleModal={toggleModal}>{children}</Modal>)

    expect(wrapper.contains(children)).toBeTruthy()

    expect(toJSON(wrapper)).toMatchSnapshot()
  })

  it('should has no animation', () => {
    const toggleModal = jest.fn()
    const children = 'test'
    const animation = 'none'

    const wrapper = shallow(
      <Modal toggleModal={toggleModal} animation={animation}>
        {children}
      </Modal>
    )

    expect(wrapper.find('.animate-none')).toHaveLength(1)

    expect(toJSON(wrapper)).toMatchSnapshot()
  })
})
