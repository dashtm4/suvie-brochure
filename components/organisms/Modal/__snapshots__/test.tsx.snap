// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Modal /> should has no animation 1`] = `
<ShadowBackdrop>
  <div
    className="relative bg-white rounded-lg shadow-2xl w-full m-auto overflow-hidden animate-none"
    style={
      Object {
        "maxWidth": "56.25rem",
      }
    }
  >
    <CloseButton
      className="absolute z-10 top-6 right-6"
      colorHex="#fff"
      handleClick={[MockFunction]}
    />
    test
  </div>
</ShadowBackdrop>
`;

exports[`<Modal /> should renders correct <Modal /> 1`] = `
<ShadowBackdrop>
  <div
    className="relative bg-white rounded-lg shadow-2xl w-full m-auto overflow-hidden animate-flip-in-x"
    style={
      Object {
        "maxWidth": "56.25rem",
      }
    }
  >
    <CloseButton
      className="absolute z-10 top-6 right-6"
      colorHex="#fff"
      handleClick={[MockFunction]}
    />
    test
  </div>
</ShadowBackdrop>
`;
